### ISSUE TITLE
Confusing warning when setting OPTIONS / name conflict with Find*.cmake files?

### ISSUE THREAD

**Issue by jhasse** [TheLartians/CPM.cmake#180]
I'm trying to add ogg like this:

```
	CPMAddPackage(NAME ogg
		URL http://downloads.xiph.org/releases/ogg/libogg-1.3.4.tar.xz
		URL_HASH SHA256=c163bc12bc300c401b6aa35907ac682671ea376f13ae0969a220f7ddf71893fe
		OPTIONS "BUILD_TESTING OFF")
```

Followed by [fetching Vorbis after that](https://github.com/jhasse/jngl/blob/5428aaa841dcc5c35237894d132e7072cc67c21a/CMakeLists.txt#L146) (using FetchContent for that now). But I'm getting the following warning:

```
CMake Warning at build/cmake/CPM_0.27.5.cmake:166 (message):
  CPM: ignoring package option for ogg: BUILD_TESTING = ON (OFF)
Call Stack (most recent call first):
  build/cmake/CPM_0.27.5.cmake:242 (CPMCheckIfPackageAlreadyAdded)
  build/CPM_modules/Findogg.cmake:2 (CPMAddPackage)
  build/_deps/vorbis-src/CMakeLists.txt:65 (find_package)


CMake Warning at build/cmake/CPM_0.27.5.cmake:166 (message):
  CPM: ignoring package option for ogg: OFF = ON ()
Call Stack (most recent call first):
  build/cmake/CPM_0.27.5.cmake:242 (CPMCheckIfPackageAlreadyAdded)
  build/CPM_modules/Findogg.cmake:2 (CPMAddPackage)
  build/_deps/vorbis-src/CMakeLists.txt:65 (find_package)
```

It seems that CPM creates a Findogg.cmake file which results in vorbis not using [their bundled FindOgg.cmake](https://github.com/xiph/vorbis/blob/3567a00101690593b3fa224bb73473162bd77dbb/cmake/FindOgg.cmake) file?

**Comment by TheLartians** [comment]
Yes, this is expected behaviour. CPM.cmake creates its own `FindXXX.cmake` modules, to allow nested dependencies that use `find_package` for dependency management. Otherwise we could accidentally add the same dependency twice (one through CPM.cmake and one from the system PM).

That warning should not appear though. The "ignoring option" warning should actually appear when adding a dependency that has already been added before, but provided with different options. As we can't add a package twice, we need to ignore any options specified later on. As package options are usually cached and unique to the dependency, there should be no name clash, however `BUILD_TESTING` is definitely too generic and perhaps confusing CPM due to a name clash. Are you changing the value of `BUILD_TESTING` somewhere else by chance?

**Comment by jhasse** [comment]
Not explicitly, but `include(CTest)` sets it, doesn't it?

**Comment by TheLartians** [comment]
Definitely [seems so](https://cmake.org/cmake/help/latest/module/CTest.html). I would actually recommend to avoid specifying project-wide options in the `OPTIONS` field of `CPMAddPackage`, as it assumes they are unique to the dependency. Perhaps use [set](https://cmake.org/cmake/help/latest/command/set.html) to change `BUILD_TESTING` to `OFF` manually before adding the dependency?

**Comment by jhasse** [comment]
But then I need to set it back to ON after adding the dependency.

**Comment by TheLartians** [comment]
If you only want to unset it for the single package, I'm afraid so. `CPMAddPackage` actually also sets the global value, as that's the expected behaviour for CMake options (in my understanding they are meant to be project-wide and specified through the command line / CMake GUI). Unfortunately, that's the way CMake options were designed so I'm not sure if there is a better way for CPM.cmake to handle them.

**Comment by iboB** [comment]
"use scoped options"  as in `OGG_BUILD_TESTING` should probably be added to the [guidelines](https://github.com/TheLartians/CPM.cmake/wiki/Preparing-projects-for-CPM.cmake)

**Comment by TheLartians** [comment]
Good point, I just updated the wiki!
